lightning/src/routing/router.rs

```diff
@@ -2144,8 +2180,9 @@ where L::Target: Logger {
 					// Verify the liquidity offered by this channel complies to the minimal contribution.
 					let contributes_sufficient_value = available_value_contribution_msat >= minimal_value_contribution_msat;
 					// Do not consider candidate hops that would exceed the maximum path length.
-					let path_length_to_node = $next_hops_path_length + 1;
-					let exceeds_max_path_length = path_length_to_node > MAX_PATH_LENGTH_ESTIMATE;
+					let intermediate_path_length_to_node = $next_hops_path_length
+						+ if $candidate.is_intermediate_hop(&payee_node_id_opt) { 1 } else { 0 };
```

Comment:
Started reviewing the PR and ran across this comment. Would this mean `targets_blinded_path` wouldn't be needed in the hop candidate structs?

Reply:
Yeah, that's right. My original approach was to limit the total number of hops (not just intermediate) per Matt's suggestion, but I ran into some issues with the way the router "counts" the number of hops in a path. Will need to look back at my notes for specifics but it would probably be a nicer API that way.